packages/fxa-shared/test/db/mysql.ts

```diff
@@ -143,4 +158,18 @@ describe('mysql', function () {
       dbOpts.connectionLimit * 2
     );
   });
+
+  it('closes connections after idling', async () => {
```

Comment:
The code is in the callback for getting a connection from the pool, so we shouldn't be getting a connection that's already been released?

> Also can we hold reference to the connection and inspect its state?

Do you mean in the tests?

Reply:
> so we shouldn't be getting a connection that's already been released?

Sorry my comment was a bit vague. You are correct the incoming connection will always be in a released state, but what is the state of the connection when the idle timeout is hit and the connection is purged? 

If the connection has already been released, and it could be since each connection [is released](https://github.com/mozilla/fxa/blob/69baebee8b3d38c444bff0fa5274f16412c0fd9f/packages/fxa-shared/db/mysql.ts#L226) after the query ends. Then when the timeout triggers you will purge an already released connection, which seems like it would result in an error as the following code paths are executed:

```
pool._purgeConnection(conn)
|-> this._removeConnection(conn)
|-> this.releaseConnection(conn)
      |-> throw new Error('Connection already released'); //Pool.js#L138
```

I have a feeling you are more familiar with the code than me, so perhaps this edge cases is bogus?

> Do you mean in the tests?

Yes in the test. Just to make sure the purge operation, which is internal, really had and continues to have the desired effect.